src/as/core/BufferGeometry.ts

```diff
@@ -150,17 +161,24 @@ export class BufferGeometry extends EventDispatcher {
 		return this
 	}
 
-	// addGroup(start: f32, count: f32, materialIndex?: f32): void {
-	// 	this.groups.push(new BufferGeometryGroup(start, count, materialIndex))
-	// 	// 	start: start,
-	// 	// 	count: count,
-	// 	// 	materialIndex: materialIndex !== undefined ? materialIndex : 0,
-	// 	// })
-	// }
+	/**
+	 * a simple helper function used to push / add new geometry groups into our array that contains
+	 * the objects that we are going to render
+	 * @param start is the beginning position we wish to offset for, use case?
+	 * @param count how big of an array we are going to create in memory
+	 * @param materialIndex the integer referenc
+	 */
+	addGroup(start: i32, count: i32, materialIndex: i32 = 0): void {
+		this.groups.push({start, count, materialIndex} as BufferGeometryGroup)
+	}
 
-	// clearGroups(): void {
-	// 	this.groups = []
-	// }
+	/**
+	 * clears buffer geometery group arrays by allocating a new empty array in place. This is required
+	 * to properly invoke the GC within the heap stack. 
+	 */
+	clearGroups(): void {
+		this.groups = new Array<BufferGeometryGroup>()
```

Comment:
Using `this.groups.length = 0` prevents making a new object (one less object to GC later), and accomplishes the same goal (release all items in the array).

Reply:
neat trick.. updat3d.